### PR TITLE
[ENH] enhancements to tag system, systematic API docs for tags

### PR DESCRIPTION
**Pull request by fkiraly**
This PR upgrades the tag system with improved inspectability, and adds systematic API docs for the tags.

The changes are as following:

* tags are now changed to be (private) first-class objects. This allows to attach more extensive documentation to tags, including examples or references. It also allows to attach tags to tags, e.g., the meta-tag fields currently in `registry._tags`, or a property such as whether a tag is meant to be user facing.
    * longer-term, it could also allow to attach validation functionality to a tag, in the form of methods, or enable dispatch, or nested dispatch.
* an API reference page for tags is added. This (ab)uses sphinx autodoc to generate tables and individual glossary pages for tags, by using the class template.

### CODE REVIEW

File: sktime/registry/_tags.py [hunk 1]
```diff
@@ -38,17 +38,243 @@
...
+
+    ``sktime`` manages objects and estimators like mini-packages,
+    with their own dependencies and compatibility requirements.
+    Dependencies are specified in the tags:
+
+    - ``"python_version"``: Python version specifier (PEP 440) for the object,
+    - ``"python_dependencies"``: list of required Python packages (PEP 440)
+    - ``"python_dependencies_alias"``: alias for package names,
+      if different from import names
+    - ``"env_marker"`: environment marker for the object (PEP 508)
```
Comment (yarnabrina) [inline sktime/registry/_tags.py:200]:
```suggestion
    - ``"env_marker"``: environment marker for the object (PEP 508)
```

File: sktime/registry/_tags.py [hunk 2]
```diff
@@ -38,17 +38,243 @@
...
```
Comment (yarnabrina) [inline sktime/registry/_tags.py:245]:
```suggestion
    - ``"env_marker"``: environment marker for the object (PEP 508)
```

**Review by yarnabrina — APPROVED** [review]
This is great. Found two missing backticks, but approved.

Another suggestion for consideration:  can we show a table of all tags (name, description, possible values, etc.) in a "Tag Overview" page, or in the bottom of "Object and estimator tags" page?

**Comment by fkiraly** [comment]
Thanks! It's not done yet though, more tags to add.
Help appreciated with that, otherwise I'd slowly add more and more.

**Comment by fkiraly** [comment]
> Another suggestion for consideration: can we show a table of all tags (name, description, possible values, etc.) in a "Tag Overview" page, or in the bottom of "Object and estimator tags" page?

Excellent idea! I was thinking additionally of an estimator type specific overview tables - like the current estimator overview, but contains only forecasters etc. For those forecasters, all user facing tags.

Should we open a new issue for these tables? We can probably implement these easily by copying the code for the current table.

**Comment by yarnabrina** [comment]
I think I've seen one of @duydl's PR where he attempted showing tag information and filtering, that'd be great to have for this then.

> It's not done yet though, more tags to add.

Are you planning to add classes for every single tag? That'd be awesome to be exhaustive, but it's likely to be very time consuming. May be we can consider first implement only the common tags (for all components/estimators) and merge, and then open this as a good first issue for new contributors?

**Comment by fkiraly** [comment]
> I think I've seen one of @duydl's PR where he attempted showing tag information and filtering, that'd be great to have for this then.

Yes - @duydl's improved table would be nicer, but as a fallback we can copy the current one (which was also fixed by him, btw).

**Comment by fkiraly** [comment]
> Are you planning to add classes for every single tag? That'd be awesome to be exhaustive, but it's likely to be very time consuming.

Less so with AI, but yes still, there's a lot of hallucination.

> May be we can consider first implement only the common tags (for all components/estimators) and merge, and then open this as a good first issue for new contributors?

Yes, that's probably a good idea.

I don't think this is a good first issue though, especially for the more obscure tags. Even though they are already described in the extension templates, I feel (a) getting this right requires strong experience in technical documentation, and (b) it's not something new contributors tend to have working experience of.

Perhaps I'm to cautious though? Let me know if you think otherwise.

**Comment by yarnabrina** [comment]
I agree with your caution completely. I've seen few messages about GSoD, not sure about the status. If that does happen, it would be appropriate IMO.

**Comment by fkiraly** [comment]
> I've seen few messages about GSoD, not sure about the status.

Oh, we didn't get it. By Apr 15, we will know the scope of donation/industry funding, and the full scope of topics. Documentation seems unlikely to be on the "funded" list though.

**Pull request merged**